### ISSUE TITLE
DialogStateManager needs a design such that the save is optimized potentially by doing it in parallel

### ISSUE THREAD

**Issue by johnataylor** [microsoft/botbuilder-dotnet#5412]

**Comment by johnataylor** [comment]
It turns out the DialogStateManager does all the saves before the saves in the bot.
It does these via saving the memoryscopes, it does not attempt to do them in parallel. And it is not clear exactly how concurrency can be introduced here.

**Comment by mrivera-ms** [comment]
This is out of scope for v4. Potentially related to #6027
